tensorflow/contrib/avro/README.md

```diff
@@ -0,0 +1,47 @@
+# Avro Record Dataset and Parser
+Why use the Avro Record Dataset and Parser?
+* You want to read your data from directly from Avro records into TensorFlow without converting it to TensorProtos first.
+* You want to read data into TensorFlow as a sparse tensor.
+
+## Installation
+
+We depend on the Avro C library, which has minimal dependencies:
+https://github.com/apache/avro/blob/master/lang/c/INSTALL
+
+A quick way to install it is as follows:
+
+```
+git clone https://github.com/apache/avro.git
+cd avro/lang/c
+mkdir -p build
+cd build
+cmake .. -DCMAKE_INSTALL_PREFIX=avrolib -DCMAKE_BUILD_TYPE=RelWithDebInfo -DTHREADSAFE=true
+make
+make test
+make install
+```
+
+Then set the environment variable AVRO_C_HOME to "<full-path-to-your-cloned-avro-repo>/lang/c/build/avrolib/"
+
+### Solutions to the limitations
+1. Add a custom option to the configure.py that asks for the path of the avroc shared library on the system
+2. Get support for cmake in bazel; which does seem to the case yet: https://github.com/bazelbuild/bazel/issues/2532.
+3. Use the cmake build chain for tensorflow -- this is experimental feature.
+4. Port the CMake build file to bazel.
+At the moment we favor option 1.
+
+## Code examples
+The code examples reside in the folder
+avro/python/examples
+To run these code examples the requirements are:
+
+* You need to install the package 'avro >= 1.7.7' in your python virtual environment that also has tensorflow installed.
```

Comment:
It seems that the "avro" pypi package is not compatible with Python 3. Can we use https://pypi.org/project/avro-python3/ instead?

Reply:
I've tried to reproduce this with python 3.6.1 and avro 1.8.2 get this error
File "/home/fraudies/pythontfavro/lib/python3.6/site-packages/avro/io.py", line 200
    bits = (((ord(self.read(1)) & 0xffL)) |
                                      ^
SyntaxError: invalid syntax

I'll evaluate the suggested dependency avro-python3.